### DISCUSSION TITLE
Samba 4.17.x on macOS is broken in `qemu` use case

### DISCUSSION THREAD

**Discussion by kjliew** [Homebrew/discussions#3868] [Everyday usage] [answered]
Samba 4.17.x on macOS is broken in `qemu` use case of mapped `\\10.0.2.4\qemu` or any shared host resources defined in `smb.conf` from system-wide smbd daemon for Guest VMs. The current solution is to roll back to the last Samba 4.16.4 or updated Formulae with Samba 4.16.6 for latest CVE fixes and pin the bottle.

This is a rare use case for `brew` version of Samba due to macOS native SMB support. I think the `brew` version of Samba would only be used specifically for QEMU virtual machines use cases. Unfortunately, such use cases are broken with Samba 4.17.x. It is broken in such a way that mapped shares are READ-ONLY and writes end up with `permission denied`.

Tested Guest VMs using QEMU v7.1.0
- Windows XP (SMB1/NTLMv2)
- ArchLinux AArch64 (SMB3_11)
- Windows 11 ARM64 (SMB3)

Samba 4.17.x is fine on Linux hosts. The same Guest VMs have no issue at writing to Samba shares hosted on ArchLinux x86_64 with the latest Samba 4.17.2.

**Comment by rimoov** [comment]


**Reply by rimoov** [reply]
Brio

**Reply by rimoov** [reply]
DOCTYPE html>
<html lang="en-us" prefix="content: http://purl.org/rss/1.0/modules/content/ dc: http://purl.org/dc/terms/ foaf: http://xmlns.com/foaf/0.1/ og: http://ogp.me/ns# rdfs: http://www.w3.org/2000/01/rdf-schema# sioc: http://rdfs.org/sioc/ns# sioct: http://rdfs.org/sioc/types# skos: http://www.w3.org/2004/02/skos/core# xsd: http://www.w3.org/2001/XMLSchema#">

**Comment by kjliew** [comment, marked as answer]
Link to upstream QEMU issue.
https://gitlab.com/qemu-project/qemu/-/issues/1299

The root cause had been identified and suggested fix is available at samba.org bugzilla. However, upstream samba rejected the fix and running smbd as non-root user is unsupported. This also explains why Samba on Linux is fine because Linux systemd run smbd as root user.

Personally, I pretty much prefer running smbd as non-root user for QEMU use case on macOS.
